src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -99,9 +99,15 @@ public T GetProperty<T>(string propertyName)
                         propertyName));
             }
 
-            return JsonConvert.DeserializeObject<T>(Properties[propertyName].SerializedValue);
+            return JsonConvert.DeserializeObject<T>(Properties[propertyName].SerializedValue, _settingsWithComprehensiveV2ContractResolver);
         }
 
+        private static readonly JsonSerializerSettings _settingsWithComprehensiveV2ContractResolver = new JsonSerializerSettings
```

Comment:
>_settingsWithComprehensiveV2ContractResolver [](start = 55, length = 44)

this is how we solve the problem of persisting SARIF types that require special serialization/deserialization to and from the property bag.